crates/fuzzing/src/oracles/dummy.rs

```diff
@@ -30,55 +26,326 @@ pub fn dummy_imports<'module>(
-pub fn dummy_value(val_ty: ValType) -> Result<Val, Trap> {
-    Ok(match val_ty {
+pub fn dummy_value(val_ty: ValType) -> Val {
+    match val_ty {
         ValType::I32 => Val::I32(0),
         ValType::I64 => Val::I64(0),
         ValType::F32 => Val::F32(0),
         ValType::F64 => Val::F64(0),
-        ValType::V128 => {
-            return Err(Trap::new(
-                "dummy_value: unsupported function return type: v128".to_string(),
-            ))
-        }
-        ValType::ExternRef => {
-            return Err(Trap::new(
-                "dummy_value: unsupported function return type: externref".to_string(),
-            ))
-        }
-        ValType::FuncRef => {
-            return Err(Trap::new(
-                "dummy_value: unsupported function return type: funcref".to_string(),
-            ))
-        }
-    })
+        ValType::V128 => Val::V128(0),
+        ValType::ExternRef => Val::ExternRef(None),
+        ValType::FuncRef => Val::FuncRef(None),
+    }
 }
 
 /// Construct a sequence of dummy values for the given types.
-pub fn dummy_values(val_tys: impl IntoIterator<Item = ValType>) -> Result<Vec<Val>, Trap> {
+pub fn dummy_values(val_tys: impl IntoIterator<Item = ValType>) -> Vec<Val> {
     val_tys.into_iter().map(dummy_value).collect()
 }
 
 /// Construct a dummy global for the given global type.
-pub fn dummy_global(store: &Store, ty: GlobalType) -> Result<Global, Trap> {
-    let val = dummy_value(ty.content().clone())?;
-    Ok(Global::new(store, ty, val).unwrap())
+pub fn dummy_global(store: &Store, ty: GlobalType) -> Global {
+    let val = dummy_value(ty.content().clone());
+    Global::new(store, ty, val).unwrap()
 }
 
 /// Construct a dummy table for the given table type.
-pub fn dummy_table(store: &Store, ty: TableType) -> Result<Table, Trap> {
-    let init_val = dummy_value(ty.element().clone())?;
-    Ok(Table::new(store, ty, init_val).unwrap())
+pub fn dummy_table(store: &Store, ty: TableType) -> Table {
+    let init_val = dummy_value(ty.element().clone());
+    Table::new(store, ty, init_val).unwrap()
 }
 
 /// Construct a dummy memory for the given memory type.
 pub fn dummy_memory(store: &Store, ty: MemoryType) -> Memory {
     Memory::new(store, ty)
 }
+
+/// Construct a dummy instance for the given instance type.
+///
+/// This is done by using the expected type to generate a module on-the-fly
+/// which we the instantiate.
+pub fn dummy_instance(store: &Store, ty: InstanceType) -> Instance {
+    let mut wat = WatGenerator::new();
+    for ty in ty.exports() {
+        wat.export(&ty);
+    }
+    let module = Module::new(store.engine(), &wat.finish()).unwrap();
+    Instance::new(store, &module, &[]).unwrap()
+}
+
+/// Construct a dummy module for the given module type.
+///
+/// This is done by using the expected type to generate a module on-the-fly.
+pub fn dummy_module(store: &Store, ty: ModuleType) -> Module {
+    let mut wat = WatGenerator::new();
```

Comment:
This doesn't seem a ton easier than using `wasm-encoder` directly. Want to file a follow up issue that we can mark good-first-issue for some contributor to replace this with `wasm-encoder`? Would help us get a little bit more fuzzing throughput here.

Reply:
I guess I can just file it, since there was nothing else for me to nitpick before this merged :-p

Reply:
Oh heh good point!